model/flow/payload.go

```diff
@@ -40,6 +42,33 @@ func (p Payload) MarshalJSON() ([]byte, error) {
 	})
 }
 
+func (p *Payload) StructureValid() error {
+	if p == nil {
+		return model.NewStructureInvalidError("nil payload")
+	}
+	for _, guarantee := range p.Guarantees {
+		if guarantee == nil {
+			return model.NewStructureInvalidError("contains nil guarantee")
+		}
+	}
+	for _, seal := range p.Seals {
+		if seal == nil {
+			return model.NewStructureInvalidError("contains nil seal")
+		}
+	}
+	for _, receipt := range p.Receipts {
+		if receipt == nil {
+			return model.NewStructureInvalidError("contains nil receipt")
+		}
+	}
+	for _, result := range p.Results {
```

Comment:
what is with the (`ChunkList`)[https://github.com/onflow/flow-go/blob/04b4bcaf206cac75e956a557a959668b100d9356/model/flow/chunk.go#L59] in the (`ExecutionResult`)[https://github.com/onflow/flow-go/blob/04b4bcaf206cac75e956a557a959668b100d9356/model/flow/execution_result.go#L15]. Elements (i.e. individual chunks) could be nil. We don't check this here. Should we?

Reply:
👍 we should